tests/unit/test_charm.py

```diff
@@ -61,431 +115,287 @@ def _read_file(path: str) -> str:
-        (root / "support/TLS/smf.pem").write_text(CERTIFICATE)
-        (root / "support/TLS/smf.csr").write_text(CSR)
-        (root / "etc/smf/uerouting.yaml").write_text(self._read_file("src/uerouting.yaml"))
-        patch_check_output.return_value = POD_IP
-        patch_nrf_url.return_value = VALID_NRF_URL
+        (root / CERTIFICATE_PATH).write_text(CERTIFICATE)
+        (root / CSR_PATH).write_text(CSR)
+        (root / UE_CONFIG_FILE_PATH).write_text(self._read_file(EXPECTED_UE_CONFIG_FILE_PATH))
         self._create_database_relation_and_populate_data()
-        self._create_nrf_relation()
-        self._create_certificates_relation()
-
-        provider_certificate = Mock(ProviderCertificate)
-        provider_certificate.certificate = CERTIFICATE
-        provider_certificate.csr = CSR
-        patch_get_assigned_certificates.return_value = [provider_certificate]
+        self.mock_get_assigned_certificates.return_value = [self._get_provider_certificate()]
 
         self.harness.set_can_connect(container=self.container_name, val=True)
 
         self.harness.charm._configure_sdcore_smf(event=Mock())
 
-        self.assertEqual(
-            (root / "etc/smf/smfcfg.yaml").read_text(),
-            self._read_file("tests/unit/expected_smfcfg.yaml"),
-        )
+        assert (root / CONFIG_FILE_PATH).read_text() == self._read_file(EXPECTED_CONFIG_FILE_PATH)
 
-    @patch("charms.sdcore_nrf_k8s.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock)
-    @patch("charm.check_output")
     def test_given_config_file_exists_and_is_not_changed_when_configure_smf_then_config_file_is_not_re_written_with_same_content(  # noqa: E501
-        self,
-        patch_check_output,
-        patch_nrf_url,
+        self, add_storage, mock_default_values, nrf_relation_id, certificates_relation_id
     ):
-        self.harness.add_storage("config", attach=True)
-        self.harness.add_storage("certs", attach=True)
         certificate = "Whatever certificate content"
         root = self.harness.get_filesystem_root(self.container_name)
-        (root / "support/TLS/smf.pem").write_text(certificate)
-        (root / "etc/smf/uerouting.yaml").write_text(self._read_file("src/uerouting.yaml"))
-        (root / "etc/smf/smfcfg.yaml").write_text(
-            self._read_file("tests/unit/expected_smfcfg.yaml")
+        (root / CERTIFICATE_PATH).write_text(certificate)
+        (root / UE_CONFIG_FILE_PATH).write_text(self._read_file(EXPECTED_UE_CONFIG_FILE_PATH))
+        (root / CONFIG_FILE_PATH).write_text(
+            self._read_file(EXPECTED_CONFIG_FILE_PATH)
         )
-        config_modification_time = (root / "etc/smf/smfcfg.yaml").stat().st_mtime
+        config_modification_time = (root / CONFIG_FILE_PATH).stat().st_mtime
         pod_ip = "1.1.1.1"
-        patch_check_output.return_value = pod_ip.encode()
-        patch_nrf_url.return_value = VALID_NRF_URL
+        self.mock_check_output.return_value = pod_ip.encode()
```

Comment:
this is already included in `mock_default_values`

Reply:
Done